extensions/eclipse-che-theia-testing-service-ext/src/browser/plugin-handle-registry.ts

```diff
@@ -0,0 +1,76 @@
+/********************************************************************************
+ * Copyright (C) 2020 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { LanguageServerAction } from './languages-test-main';
+import { LanguagesExt } from '@theia/plugin-ext';
+
+/**
+ * This class keeps a registry of which plugins map to which handle
+ */
+
+export interface LanguageFeatureRegistration {
+    languagesExtImpl: LanguagesExt; // The languagesMain that registered this plugin in the registry
+    providerHandles: Map<LanguageServerAction, number>; // A map of language actions to their handle
+}
+
+export interface LanguagesMainHandle {
+    handle: number;
+    languagesExt: LanguagesExt;
+}
+
+@injectable()
+export class PluginHandleRegistry {
+    pluginRegistrationMap: Map<string, LanguageFeatureRegistration> = new Map();
+
+    private sleep(ms: number) {
```

Comment:
Shouldn't we resolve a promise when language feature is there instead of having waiting loops ?

code is doing: 
wait for 60s with small loops of 100ms to get something and use that object

while maybe it should do:
call something after language is ready.(callback when language is ready) (no wait loop)